### ISSUE TITLE
Set lock screen?

### ISSUE THREAD

**Issue by lucioflaviodev** [liuhong1happy/react-native-wallpaper-manager#7]
I tried using this:
`wallpaperManager.setBitmap (bitmap, null, true, WallpaperManager.FLAG_SYSTEM | WallpaperManager.FLAG_LOCK);`

instead of:
`wallpaperManager.setBitmap (bitmap);`

but did not work, presented this error:

> can not find symbol ...
> variable: variable FLAG_SYSTEM
> location: class WallpaperManager

Can someone help?

**Comment by harshmewada** [comment]
Same problem here,

**Comment by ZainabShakeel** [comment]
any solution found for this ? my image is not centered when set as Wallpaper

**Comment by harshmewada** [comment]
I also noticed my wallpapers are also not centered. I thought of using android native intents for setting wallpapers but i don't know java. I want to solve this problem as soon as possible. If you are interested, let's connect on chat and try to solve the problem of setting wallpapers in react native as soon as possible.

**Comment by ZainabShakeel** [comment]
yes i want to solve this issue beacuse there is no method in react-native-wallpaper-manager to resize an image.

**Comment by ZainabShakeel** [comment]
@harshmewada just remove the all the  .centerCrop() from wallPapermanager.java which is in node_modules\react-native-wallpaper-manager\android\src\main\java\com\cunyutech\hollyliu\reactnative\wallpaper\WallPaperManager.java  and the wallpaper would be diaplayed fine

**Comment by ZainabShakeel** [comment]
@harshmewada and line # 222 in wallpapermanager.java i change this " return new SimpleTarget<byte[]>(1080, 1920)" to " return new SimpleTarget<byte[]>(700, 1100)"

**Comment by harshmewada** [comment]
> @harshmewada and line # 222 in wallpapermanager.java i change this " return new SimpleTarget<byte[]>(1080, 1920)" to " return new SimpleTarget<byte[]>(700, 1100)"

Thank you for this solution. My app is currently published on play store and i have overcomed this issue by cropping images to device resolution and then passing them to the function. So that on each and every phone, the images are centered. My main problem is setting lockscreen wallpapers.
I don't know much about java, and also i don't know where to put the FLAG_LOCK in the code. if you know where to put this line and if you succeed, please let me know.

FLAG_LOCK
--




FLAG_LOCK
--




FLAG_LOCK
--

**Comment by harshmewada** [comment]
i know we can somehow pass uri of image to this below given intent so that we can pass our image to built in wallpaper chooser functionality. I just don't know how to implement this with react-native.

    Intent intent = new Intent(Intent.ACTION_ATTACH_DATA);
    intent.addCategory(Intent.CATEGORY_DEFAULT);
    intent.setDataAndType(uri, "image/jpeg");
    intent.putExtra("mimeType", "image/jpeg");
    this.startActivity(Intent.createChooser(intent, "Set as:"));

if somehow anyone can implement this, then almost all of our problems get solved

**Comment by oknust** [comment]
> > @harshmewada and line # 222 in wallpapermanager.java i change this " return new SimpleTarget<byte[]>(1080, 1920)" to " return new SimpleTarget<byte[]>(700, 1100)"
> 
> Thank you for this solution. My app is currently published on play store and i have overcomed this issue by cropping images to device resolution and then passing them to the function. So that on each and every phone, the images are centered. My main problem is setting lockscreen wallpapers.
> I don't know much about java, and also i don't know where to put the FLAG_LOCK in the code. if you know where to put this line and if you succeed, please let me know.
> 
> ## FLAG_LOCK
> ## FLAG_LOCK
> ## FLAG_LOCK

Hi @harshmewada,
How did you crop images to device resolution ?  Please help me :)

**Comment by ibelgin** [comment]
> i know we can somehow pass uri of image to this below given intent so that we can pass our image to built in wallpaper chooser functionality. I just don't know how to implement this with react-native.
> 
> ```
> Intent intent = new Intent(Intent.ACTION_ATTACH_DATA);
> intent.addCategory(Intent.CATEGORY_DEFAULT);
> intent.setDataAndType(uri, "image/jpeg");
> intent.putExtra("mimeType", "image/jpeg");
> this.startActivity(Intent.createChooser(intent, "Set as:"));
> ```
> 
> if somehow anyone can implement this, then almost all of our problems get solved

Any way to convert this to react Native.... I need an module or an easy to set wallpaper

**Comment by usmanrwp** [comment]
did you find any solution?

**Comment by harshmewada** [comment]
https://www.npmjs.com/package/react-native-manage-wallpaper


This module works for me

**Comment by usmanrwp** [comment]
can you please share your code ? because my app crash. when i can the function to set wallpaper. Thank You

**Comment by usmanrwp** [comment]
can you please share your code?

**Comment by natanfeitosa** [comment]
I think this project is abandoned. You can use [this lib](https://github.com/natanfeitosa/rn-wallpapers) that was released a few days ago
